File: examples/javascript/matrix.js
import { OSRMPlus } from "osrmplus";

const client = new OSRMPlus("your_api_key");

const result = await client.matrix({
  coordinates: [
    [74.3436, 31.5497], // Gulberg
    [74.265, 31.47],    // Model Town
    [74.3, 31.5],       // Liberty
    [74.28, 31.52],     // Johar Town
  ],
  annotations: "duration,distance",
});

console.log("Duration matrix (seconds):");
console.table(result.durations);

console.log("Distance matrix (meters):");
console.table(result.distances);

File: examples/python/matrix.py
"""Compute a distance matrix between multiple locations."""

from osrmplus import Client

client = Client("your_api_key")

locations = [
    (74.3436, 31.5497),  # Gulberg
    (74.2650, 31.4700),  # Model Town
    (74.3000, 31.5000),  # Liberty
    (74.2800, 31.5200),  # Johar Town
]

result = client.matrix(
    coordinates=locations,
    annotations="duration,distance",
)

print("Duration matrix (seconds):")
for row in result["durations"]:
    print([f"{v:.0f}" if v else "---" for v in row])

print("\nDistance matrix (meters):")
for row in result["distances"]:
    print([f"{v:.0f}" if v else "---" for v in row])
